### PR TITLE
Fix kaetzchen plumbing

### PR DESCRIPTION
**Pull request by david415**
Reopening PR. DO NOT MERGE. This branch is for review purposes only. Once approved I will make a new branch with fewer commits.

### CODE REVIEW

**Comment by david415** [comment]
@Yawning ok i've added unit tests for this new feature. please do let me know if there are things you would like me to change with any of this. for now let's use this branch for code review and i'll clean up the commit messages after i get your approval.

File: internal/provider/kaetzchen/kaetzchen.go [hunk 1]
```diff
@@ -80,3 +87,198 @@ var BuiltInCtors = map[string]BuiltInCtorFn{
...
+		var pkt *packet.Packet
+		select {
+		case <-k.HaltCh():
+			k.log.Debugf("Terminating gracefully.")
+			return
+		case e := <-ch:
+			pkt = e.(*packet.Packet)
+			if dwellTime := monotime.Now() - pkt.DispatchAt; dwellTime > maxDwell {
+				count := k.incrementDropCounter()
+				k.log.Debugf("Dropping packet: %v (Spend %v in queue), total drops %d", pkt.ID, dwellTime, count)
```
Comment (mixmasala) [inline internal/provider/kaetzchen/kaetzchen.go]: would it be useful to keep the fraction of dropped packets to forwarded packets?

File: internal/provider/kaetzchen/kaetzchen_test.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+				},
+			},
+		},
+	}
+
+	kaetzWorker, err := New(goo)
+	require.NoError(err)
+
+	params := make(Parameters)
+	params[ParameterEndpoint] = "test"
```
Comment (mixmasala) [inline internal/provider/kaetzchen/kaetzchen_test.go]: might it make sense to have test cases for invalid kaetzchen names?

Comment (david415) [reply internal/provider/kaetzchen/kaetzchen_test.go]: how is it invalid? i mean... the spec says they should start with "+". But other than that... here we define and register a new kaetzchen service.

File: internal/provider/kaetzchen/kaetzchen.go (hunk 1 above)
Comment (david415) [reply internal/provider/kaetzchen/kaetzchen.go]: I don't have any immediate use for stats collection. I was simply keeping a counter to facilitate writing the tests. We do have a PR that I closed... maybe you'd like to look at it... it add some prometheus counters and it's web listener/stats aggregator which is probed by the UI to gather stats.

**Comment by david415** [comment]
https://github.com/katzenpost/server/pull/59

**Pull request closed without merging**